### DISCUSSION TITLE
Inkscape-figure manager fails to build .pdf and .pdf_tex files

### DISCUSSION THREAD

**Discussion by inctechs** [sleepymalc/VSCode-LaTeX-Inkscape#14] [Q&A] [answered]
Hi,

I apologise if this is the wrong place for this question. I am just not very familiar with a proper way of mentioning coding problems.

The issue me and my friends encounter is the following: Whenever we create a new figure ("insert" mode + "ctrl+f") the Inkscape figure manager function "create" seems to work perfectly. We get the .svg file in the ./figures/ file as intended. But what is missing are the .pdf and .pdf_tex files. We have to manually save the file as .pdf_tex and .pdf and then everything works just fine. Hence, we assume an issue with the file manager setting up a "file watcher" so that when hitting "ctrl-s" the Inkscape figure manager fails to also save the file as pdf+latex.

If you need anything more to describe the problem, please just let me know.
Since three different people who installed everything individually came across the same problem, I thought it might be good to ask here.

Thanks for your time and help.

**Comment by sleepymalc** [comment]
What's the command line output after you hit `ctrl`+`f` in `VISUAL` mode? You'll need to set up the file-watcher before you start saving changes.

**Reply by inctechs** [reply]
The runCommand terminal gives this sequence of errors:
`(org.inkscape.Inkscape:20083): GLib-CRITICAL **: 10:30:05.536: g_date_time_format: assertion 'datetime != NULL' failed`

`(org.inkscape.Inkscape:20083): GLib-CRITICAL **: 10:30:05.537: g_date_time_format: assertion 'datetime != NULL' failed`

`(org.inkscape.Inkscape:20083): GLib-CRITICAL **: 10:30:05.537: g_date_time_format: assertion 'datetime != NULL' failed`

`(org.inkscape.Inkscape:20083): GLib-CRITICAL **: 10:30:05.538: g_date_time_format: assertion 'datetime != NULL' failed`

`(org.inkscape.Inkscape:20083): GLib-CRITICAL **: 10:30:05.539: g_date_time_format: assertion 'datetime != NULL' failed`

`(org.inkscape.Inkscape:20083): GLib-CRITICAL **: 10:30:05.540: g_date_time_format: assertion 'datetime != NULL' failed
Shortcut::add_shortcut: duplicate shortcut found for: <Primary>comma  Old: app.preferences  New: app.transform-scale(0.5) !`

But this is the only feedback I get after hitting `ctrl+f` in `Visual` mode.
Setting up the file watcher I followed your instructions and before creating any picture the file watcher was returning nothing as expected:

` /test_git_overleaf_snippet % inkscape-figures watch`
`/test_git_overleaf_snippet % `

The above gives the result when using the file watcher without any images in the file.

But the error code seems to be different, because the last time I tried to do it I got the error code '15004'.

**Reply by sleepymalc** [reply]
I think you should first manually test the three functionalities provided by `inkscape-file-manager`: `create`, `edit`, `watch`, and see whether they work as expected. Then you set up keybinding with `runcommands` and use VS Code to trigger them.

Personally, I can't recall whether I have seen those errors or not, but I think the problem may be the fact that you replace the wrong `inkscape-file-manager`'s used by the system if your python environment is complicated. Try to identify all the folders of `inkscape-file-manager` and tweak them manually. Sorry for the inconvenience but I have no further ideas...

**Reply by inctechs** [reply]
No absolutely no need to apologise, thank you very much for your time and help. I will try to figure it out with your hints and mention a solution here if I find one. Thanks again.

**Reply by sleepymalc** [reply]
sure, update the progress and let me know if there's a further problem and I'm willing to help.

**Comment by Francesco-2003** [comment]
Hi there, I have the same problem; has it been solved?

**Reply by sleepymalc** [reply]
It may be a good idea to update everything in `brew` by `brew upgrade`...

**Comment by ColtWhitley** [comment]
Any updates on this problem? I am running into the same errors

**Comment by sleepymalc** [comment, marked as answer]
Maybe tried `brew install fswatch`? I found this [here](https://github.com/gillescastel/inkscape-figures/issues/31).
